staging/src/k8s.io/kubectl/pkg/scale/scale.go

```diff
@@ -113,23 +113,25 @@ type genericScaler struct {
 var _ Scaler = &genericScaler{}
 
 // ScaleSimple updates a scale of a given resource. It returns the resourceVersion of the scale if the update was successful.
-func (s *genericScaler) ScaleSimple(namespace, name string, preconditions *ScalePrecondition, newSize uint, gr schema.GroupResource) (updatedResourceVersion string, err error) {
-	scale := &autoscalingv1.Scale{
-		ObjectMeta: metav1.ObjectMeta{Namespace: namespace, Name: name},
-	}
+func (s *genericScaler) ScaleSimple(namespace, name string, preconditions *ScalePrecondition, newSize uint, gvr schema.GroupVersionResource) (updatedResourceVersion string, err error) {
+	var (
+		updatedScale *autoscalingv1.Scale
+	)
 	if preconditions != nil {
-		var err error
-		scale, err = s.scaleNamespacer.Scales(namespace).Get(gr, name)
+		scale, err := s.scaleNamespacer.Scales(namespace).Get(gvr.GroupResource(), name)
 		if err != nil {
 			return "", err
 		}
-		if err := preconditions.validate(scale); err != nil {
+		if err = preconditions.validate(scale); err != nil {
 			return "", err
 		}
+		scale.Spec.Replicas = int32(newSize)
+		updatedScale, err = s.scaleNamespacer.Scales(namespace).Update(gvr.GroupResource(), scale)
+	} else {
+		patch := []byte(fmt.Sprintf(`{"spec":{"replicas":%d}}`, newSize))
+		updatedScale, err = s.scaleNamespacer.Scales(namespace).Patch(gvr, name, types.MergePatchType, patch)
 	}
 
-	scale.Spec.Replicas = int32(newSize)
-	updatedScale, err := s.scaleNamespacer.Scales(namespace).Update(gr, scale)
 	if err != nil {
```

Comment:
You could have just kept the updatedScale inside the if block and return it from there, I think it'd have been cleaner, but that's just really taste so don't worry :-)